chall01/mlindhol.py

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python
+import sys
+
+
+def error():
+    print("usage: ./mlindhol.py <a-zA-Z string>")
+    exit(-1)
+
+
+if len(sys.argv) != 2 or len(sys.argv[1]) == 0:
+    error()
+
+morse_array = [".-", "-...", "-.-.", "-..", ".", "..-.", "--.", "....", "..", ".---", "-.-", ".-..", "--",
+               "-.", "---", ".--.", "--.-", ".-.", "...", "-", "..-", "...-", ".--", "-..-", "-.--", "--.."]
+
+input_str = sys.argv[1]
+
+for x in input_str:
+    av = ord(x)
+    if av < 32 or 32 < av < 65 or 90 < av < 97 or av > 122:
+        error()
+
+for s in input_str:
+    if s is " ":
+        print(" ", end="")
+    else:
+        y = 64 if s.isupper() else 96
+        letter = ord(s) - y
+        print(morse_array[letter - 1], end="")
```

Comment:
would be better avoiding printing char by char but to append into a string and then print the string at the end.

and

A good practice is to protect your python scripts that are meant to be run from running on import.
It's done by
```python
if __name__ == "__main__":
```
and then your code (or even better would be to put your code to function called main() and call it here)